cirq/circuits/circuit_test.py

```diff
@@ -345,43 +345,227 @@ def test_append_strategies():
     ])
 
 
-def test_insert():
-    a = cirq.QubitId()
-    b = cirq.QubitId()
+def test_insert_op_tree_new():
```

Comment:
Perhaps a simpler way to test this would be to have a list of `(given_index, actual_index, operation)` and to confirm that inserting with the given index was equivalent to inserting with the actual index. You could then have a parameterized test based on all four insertion strategies, without having to figure out the exact circuits manually.